components/dynamic/SummitData/Weather/index.js

```diff
@@ -0,0 +1,51 @@
+import Loader from "@/components/atomic/Loader";
+import PrecipitationCurrent from "@/components/widgets/PrecipitationCurrent";
+import TemperatureCurrent from "@/components/widgets/TemperatureCurrent";
+import { useSummitData } from "@/contexts/SummitData";
+import { defaultUnits, useWeatherUnit } from "@/contexts/WeatherUnit";
+import convert from "convert";
+
+const Weather = () => {
+  const { tempUnit, windspeedUnit } = useWeatherUnit();
+  const { currentData: data, loading = true } = useSummitData();
+
+  if (loading && !data) return <Loader isVisible={true} />;
```

Comment:
since `loading` won't go to `false` until all complete, we'll need to check in each container if the provider is loading and the data hasn't been set yet. Or we could go to individualized loading statuses but I think this is ok as-is